lib/phoenix_live_view.ex

```diff
@@ -761,10 +761,10 @@ defmodule Phoenix.LiveView do
     end
   end
 
-  def assign_new(%{__changed__: changed} = assigns, key, fun) when is_function(fun, 0) do
+  def assign_new(%{__changed__: changed} = assigns, key, fun) when is_function(fun, 1) do
```

Comment:
Hi @joshchernoff unfortunately this line is introducing a breaking change, so I believe what you want is to actually add a new function clause like you did `assign_new(...) when is_function(fun, 1)` and keep the current one unchanged, in other words have support for both. Also update docs to explain and add an example of this new feature :)

And considering that test/phoenix_live_view_test.exs have the new test "has access to new assigns", then the other changes on that file and also the change on test/phoenix_component_test.exs may not be needed.